### PR TITLE
Toz activation token - version 2

### PR DESCRIPTION
**Pull request by ppiotr191**
- send register token
- create and activate user by token and proposal

### CODE REVIEW

File: src/main/java/com/intive/patronage/toz/config/ApiUrl.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
     public final static String ORGANIZATION_INFO_PATH = "/organization/info";
     public final static String TOKENS_PATH = "/tokens";
     public final static String ACTIVATION_PATH = "/activate";
+    public final static String REGISTER_PATH = "/register";
```
Comment (mlotysz) [inline src/main/java/com/intive/patronage/toz/config/ApiUrl.java]: I think it belong to `/users` resource, hence I propose: `/users/register`

File: src/main/java/com/intive/patronage/toz/users/UserActivationService.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        this.jwtFactory = jwtFactory;
+        this.userRepository = userRepository;
+        this.proposalRepository = proposalRepository;
+        this.mailTemplatesService = mailTemplatesService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public void sendActivationEmail(Proposal proposal) throws IOException, MessagingException {
+
+        System.out.println(proposal);
```
Comment (mlotysz) [inline src/main/java/com/intive/patronage/toz/users/UserActivationService.java]: Remove.

File: src/main/java/com/intive/patronage/toz/users/model/db/User.java [hunk 3]
```diff
@@ -19,10 +19,12 @@
 public class User extends PersonalData {
 
     private String passwordHash;
+    private boolean isActive;
 
     @Temporal(TemporalType.TIMESTAMP)
     private Date passwordChangeDate;
 
+
```
Comment (mlotysz) [inline src/main/java/com/intive/patronage/toz/users/model/db/User.java]: Remove.

File: src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        userActivationService = new UserActivationService(
+                jwtFactory,
+                userRepository,
+                proposalRepository,
+                jwtParser,
+                mailService,
+                mailTemplatesService,
+                passwordEncoder
+        );
+
```
Comment (mlotysz) [inline src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java]: Remove extra empty lines. One is enough.

File: src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.MessageSource;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
```
Comment (mlotysz) [inline src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java]: Reorganize imports.

File: src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+import java.security.SecureRandom;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.any;
```
Comment (mlotysz) [inline src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java]: Add empty line - reformat code?

File: src/test/java/com/intive/patronage/toz/users/UserTestsUtils.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.intive.patronage.toz.users;
+
+import com.intive.patronage.toz.users.model.db.Role;
+import com.intive.patronage.toz.users.model.db.User;
+
+import java.util.Collections;
+
+public class UserTestsUtils {
```
Comment (mlotysz) [inline src/test/java/com/intive/patronage/toz/users/UserTestsUtils.java]: Could this class be removed? I'll move `getUserWithRole` to `UserDataProvider` class.

File: src/main/java/com/intive/patronage/toz/users/model/db/User.java [hunk 8]
```diff
@@ -19,10 +19,12 @@
 public class User extends PersonalData {
 
     private String passwordHash;
+    private boolean isActive;
```
Comment (mlotysz) [inline src/main/java/com/intive/patronage/toz/users/model/db/User.java]: Is this flag used somewhere? It's defaults to `false` and what then: "non active" user could use our API still?

File: src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java (hunk 5 above)
Comment (ppiotr191) [inline src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java]: Can you describe, what do you mean? Remove empty lines?

Comment (mlotysz) [inline src/test/java/com/intive/patronage/toz/users/UserActivationServiceTest.java]: Yes, please.

File: src/test/java/com/intive/patronage/toz/users/UserTestsUtils.java (hunk 7 above)
Comment (ppiotr191) [inline src/test/java/com/intive/patronage/toz/users/UserTestsUtils.java]: removed

**Pull request merged**